### PR TITLE
feat(react): automatically detect npm dependencies as external when building library

### PR DESCRIPTION
**Pull request by jaysoo**
This MR allows React/Web/Next libraries to be built without manually adding entries to `external` option.

## Current Behavior

If you used say 'lodash' you need  to use option `"external": ["lodash"]` in project configuration.

## Expected Behavior

Npm packages should not need  manual  `external` entry. Nx already knows this information and can provide it automatically.

### CODE REVIEW

File: packages/web/src/executors/package/package.impl.ts [hunk 1]
```diff
@@ -39,6 +28,16 @@ import {
 import { deleteOutputDir } from '../../utils/delete-output-dir';
 import { runRollup } from './run-rollup';
 
+// These use require because the ES import isn't correct.
+const resolve = require('@rollup/plugin-node-resolve');
```
Comment (ZackDeRose) [inline packages/web/src/executors/package/package.impl.ts:32]: Question: can we not use the import syntax for importing a default export here? e.g.:

```ts
import resolve from '@rollup/plugin-node-resolve';
```

**Review by ZackDeRose — APPROVED** [review]
LGTM

**Review by cyrus-za — COMMENTED** [review]
This looks awesome. Thanks so much. Will this work with pnpm? When I tried your gist, my `id` value was the full system path

File: packages/web/src/executors/package/package.impl.ts (hunk 1 above)
Comment (jaysoo) [reply packages/web/src/executors/package/package.impl.ts:32]: It depends on the es interopt option in tsconfig. At the very least we can probably do `import resolve = require('@rollup/plugin-node-resolve');`

**Comment by jaysoo** [comment]
> This looks awesome. Thanks so much. Will this work with pnpm? When I tried your gist, my `id` value was the full system path

Yes there isn't any different between how we handle npm, yarn,  pnpm from a dependency detection perspective.

File: packages/web/src/executors/package/package.impl.ts (hunk 1 above)
Comment (cyrus-za) [reply packages/web/src/executors/package/package.impl.ts:32]: > > This looks awesome. Thanks so much. Will this work with pnpm? When I tried your gist, my `id` value was the full system path
> 
> Yes there isn't any different between how we handle npm, yarn, pnpm from a dependency detection perspective.

Then I am not sure what I am doing wrong but I am getting errors 
```sh
Error during bundle: Error: 'isEmpty' is not exported by node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/fp.js
Bundle failed: ui-hooks-use-resize-observer
```

`isEmpty` is imported from lodash in this case. 

```json
{
  "root": "libs/ui",
  "sourceRoot": "libs/ui",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/web:package",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ui/package",
        "tsConfig": "libs/ui/tsconfig.lib.json",
        "project": "libs/ui/package.json",
        "entryFile": "libs/ui/index.ts",
        "assets": [
          {
            "glob": "libs/ui/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "release": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx semantic-release"
          }
        ],
        "cwd": "libs/ui"
      }
    }
  },
  "tags": []
}

```

**Comment by cyrus-za** [comment]
> > This looks awesome. Thanks so much. Will this work with pnpm? When I tried your gist, my `id` value was the full system path
> 
> Yes there isn't any different between how we handle npm, yarn, pnpm from a dependency detection perspective.

Then I am not sure what I am doing wrong but I am getting errors 
```sh
Error during bundle: Error: 'isEmpty' is not exported by node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/fp.js
Bundle failed: ui-hooks-use-resize-observer
```

`isEmpty` is imported from lodash in this case. 

```json
{
  "root": "libs/ui",
  "sourceRoot": "libs/ui",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/web:package",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ui/package",
        "tsConfig": "libs/ui/tsconfig.lib.json",
        "project": "libs/ui/package.json",
        "entryFile": "libs/ui/index.ts",
        "assets": [
          {
            "glob": "libs/ui/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "release": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npx semantic-release"
          }
        ],
        "cwd": "libs/ui"
      }
    }
  },
  "tags": []
}

```

**Pull request merged**